### PR TITLE
Drop class name analogies favoring technical names

### PR DESCRIPTION
**Pull request by rvhonorato**
This is a boring PR to make the code more accessible.

I do believe analogy is a great technique to match unrelated objects in order facilitate understanding and it is seen in many large projects out there in the wild. However me and @joaomcteixeira are getting lost in the ones that were present here since they were often actually quite metaphoric. 

Additionally, since we aim the project to have a wide reach, we need to also consider code accessibility and making it more boring can increase its readability. There will still be plenty of room for creative expression later on :)

With this PR I also linted to code to comply with [E501](https://www.flake8rules.com/rules/E501.html).
In this branch there are no tests yet, so I tested it by running the examples and checking if things were working.

### CODE REVIEW

File: haddock/defaults.py [hunk 1]
```diff
@@ -37,57 +38,108 @@ class Default:
...
-        "trans_vector_42": data_path / "toppar/initial_positions/trans_vector_42",
-        "trans_vector_43": data_path / "toppar/initial_positions/trans_vector_43",
-        "trans_vector_44": data_path / "toppar/initial_positions/trans_vector_44",
-        "trans_vector_45": data_path / "toppar/initial_positions/trans_vector_45",
-        "trans_vector_46": data_path / "toppar/initial_positions/trans_vector_46",
-        "trans_vector_47": data_path / "toppar/initial_positions/trans_vector_47",
-        "trans_vector_48": data_path / "toppar/initial_positions/trans_vector_48",
-        "trans_vector_49": data_path / "toppar/initial_positions/trans_vector_49",
-        "trans_vector_50": data_path / "toppar/initial_positions/trans_vector_50"
+        "trans_vector_0": (data_path /
```
Comment (brianjimenez) [inline haddock/defaults.py:41]: This is a classic example where I would not personally follow strict 80-lenght line except if committing a major number of lines is intended in order to inflate statistics.

File: haddock/modules/sampling/lightdock.py [hunk 2]
```diff
@@ -27,34 +27,43 @@ def run(self, module_information):
         self.patch_defaults(module_information)
 
         # Get the models generated in previous step
-        models_to_score = [p for p in self.previous_io.output if p.file_type == Format.PDB]
```
Comment (brianjimenez) [inline haddock/modules/sampling/lightdock.py:30]: List comprehension is faster and IMHO more readable here.

File: haddock/modules/topology/default.py [hunk 3]
```diff
@@ -56,33 +56,50 @@ def run(self, module_information):
...
             logger.info("CNS engine has finished")
 
-            # Check for generated output, fail it not all expected files are found
+            # Check for generated output, fail it not all expected files
+            #  are found
             expected = []
             not_found = []
             for model in models:
                 model_name = model.stem
-                if not (processed_pdb := self.path / f"{model_name}_haddock.{Format.PDB}").is_file():
```
Comment (brianjimenez) [inline haddock/modules/topology/default.py:77]: Why this? Using modern Python and walrus operator is not a crime...

File: haddock/ontology.py [hunk 4]
```diff
@@ -29,7 +30,9 @@ def __init__(self, file_name, file_type, path='.'):
         self.path = str(Path(path).parent.absolute())
 
     def __repr__(self):
-        return f"[{self.file_type}|{self.created}] {Path(self.path) / self.file_name}"
+        rep = (f"[{self.file_type}|{self.created}] "
```
Comment (brianjimenez) [inline haddock/ontology.py:33]: Less efficient and unnecessary.

File: haddock/workflow.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        # Create a recipe from a TOML file
+        self.recipe = Recipe(path)
+
+    def run(self):
+        """High level workflow composer"""
+        for step in self.recipe.steps[self.start:]:
+            step.execute()
+
+
+class Recipe:
```
Comment (brianjimenez) [inline haddock/workflow.py:33]: If not using metaphors anymore, maybe there is a better name for this class too?

**Review by brianjimenez — COMMENTED** [review]
Boring review too. I am not a fan of strict linting, specially of line-length rules, but your call. List comprehensions and walrus operator are more pythonic options though.

File: haddock/modules/scoring/default.py [hunk 6]
```diff
@@ -51,11 +56,21 @@ def run(self, module_information):
         jobs = []
 
         # Get the models generated in previous step
-        models_to_score = [p for p in self.previous_io.output if p.file_type == Format.PDB]
+        models_to_score = []
```
Comment (joaomcteixeira) [inline haddock/modules/scoring/default.py:59]: same as before.
if you want to be purist, these logic should be some kind of method of function. but for now just keep the list comprehension

**Review by joaomcteixeira — COMMENTED** [review]
Just made general comments. Nothing critical. Somethings I personally dislike, but they are not mistakes either.

File: haddock/workflow.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+                    self.steps.append(Step(stage,
+                                           num_stage,
+                                           content["stage"][stage],
+                                           recipe_path.parent))
+            except RecipeError as re:
+                logger.error(f"Error found while parsing course {stage}")
+                raise HaddockError from re
+
+
+class Step:
```
Comment (joaomcteixeira) [inline haddock/workflow.py:67]: Needs documentation

**Comment by joaomcteixeira** [comment]
However, despite we change the class names to more technical ones we should keep the nice HADDOCK-related interface for the users. Phrases of sailors are nice :boat:

File: haddock/defaults.py (hunk 1 above)
Comment (rvhonorato) [reply haddock/defaults.py:41]: ooh this is great @joaomcteixeira, thanks!

**Pull request merged**